specs/PushNotifications/PushNotifications-spec.md

```diff
@@ -0,0 +1,400 @@
+      </com:ExeServer>
+    </com:ComServer>
+  </com:Extension>
+</Extensions>
+```
+
+## UWP InProc activator
+The UWP inproc activator flow differs from the Win32 inproc activator flow since there is technically no WinMain activation associated with UWPs. Instead the App will have to  override and implement OnBackgroundActivated which is tightly coupled to Windows.UI.Xaml workstreams.
+
+# API Details
+```c# (but really MIDL3)
+
+namespace Microsoft.ProjectReunion.PushNotifications
+{
+    // Event args in WinMain activation payload.
+    runtimeclass PushActivatedEventArgs
+    {
+        // Initialize using the raw byte payload
+        PushActivatedEventArgs(Windows.ApplicationModel.Background.IBackgroundTaskInstance backgroundTask);
+
+        // The Push payload
+        byte[] Payload { get; };
+
+        // Takes a deferral to run under specific modes like low power mode
+        void GetDeferral();
+
+        // Closes the Deferral taken on a Push Trigger. If this is not called by the developer, the deferral will automatically be invoked on destruction.
+        void CompleteDeferral();
+
+        // Subscribe to Cancelled event handler to be signalled when resource policies are no longer true like 30s wallclock timer
+        event Windows.ApplicationModel.Background.BackgroundTaskCanceledEventHandler Canceled;
+
+        // The number of times Resource management suspended the Task
+        UInt32 SuspendedCount { get; };
+    };
+
+    // An abstraction over the inproc activation Registration flow
+    runtimeclass InProcActivatorDetails
+    {
+        // Initialize using a manifest defined activatorId to Activate the InProc COM server
+        InProcActivatorDetails(Guid taskClsid);
+
+        // Initialize an inproc activator for LOW-IL apps like UWP with no COM Server specified in manifest
+        InProcActivatorDetails();
+
+        // The CLSID associated with the Client COM server that reunion will adopt
+        Guid TaskClsid{ get; };
+
+        // The conditions under which Push Triggers would execute
+        Windows.ApplicationModel.Background.IBackgroundCondition Condition;
+
+        // Event handler to subscribe to if the app is in the foreground
+        event Windows.Foundation.EventHandler<PushActivatedEventArgs> PushActivated;
+
+        // The backgroundRegistration that gets populated on successful registration
+        Windows.ApplicationModel.Background.IBackgroundTaskRegistration Registration;
+    };
+
+    enum ChannelStatus
+    {
+        InProgress, // The request is in progress and there is no retry operation
+        InProgressRetry, // The request is in progress and is in a backoff retry state. Check ExtendedError for HRESULT for retryable error.
+        CompletedSuccess, // The request completed successfully
+        CompletedFailure, // The request failed with some critical internal error. Check ExtendedError for HRESULT
+    };
+
+    runtimeclass ChannelResult
+    {
+        ChannelResult(
```

Comment:
DISCUSSION: Start of a pattern - allowing apps to create types they can use for testing or driving their own state machines.